sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BulkExecutionOptions.java

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.ImplementationBridgeHelpers;
+import com.azure.cosmos.implementation.batch.BatchRequestResponseConstants;
+import com.azure.cosmos.implementation.spark.OperationContextAndListenerTuple;
+import com.azure.cosmos.util.Beta;
+
+import java.time.Duration;
+
+public final class BulkExecutionOptions {
+    private int maxMicroBatchSize = BatchRequestResponseConstants.MAX_OPERATIONS_IN_DIRECT_MODE_BATCH_REQUEST;
+    private int maxMicroBatchConcurrency = BatchRequestResponseConstants.DEFAULT_MAX_MICRO_BATCH_CONCURRENCY;
+    private double maxMicroBatchRetryRate = BatchRequestResponseConstants.DEFAULT_MAX_MICRO_BATCH_RETRY_RATE;
+    private double minMicroBatchRetryRate = BatchRequestResponseConstants.DEFAULT_MIN_MICRO_BATCH_RETRY_RATE;
+    private Duration maxMicroBatchInterval = Duration.ofMillis(
+        BatchRequestResponseConstants.DEFAULT_MAX_MICRO_BATCH_INTERVAL_IN_MILLISECONDS);
+    private final Object legacyBatchScopedContext;
+    private final BulkExecutionThresholds thresholds;
+    private OperationContextAndListenerTuple operationContextAndListenerTuple;
+
+    /**
+     * Constructor
+     * @param thresholds thresholds
+     */
+    BulkExecutionOptions(Object legacyBatchScopedContext, BulkExecutionThresholds thresholds) {
+        this.legacyBatchScopedContext = legacyBatchScopedContext;
+        if (thresholds == null) {
+            this.thresholds = new BulkExecutionThresholds();
+        } else {
+            this.thresholds = thresholds;
+        }
+    }
+
+    /**
+     * Constructor
+     * @param thresholds thresholds
+     */
+    @Beta(value = Beta.SinceVersion.V4_18_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public BulkExecutionOptions(BulkExecutionThresholds thresholds) {
+        this(null, thresholds);
+    }
+
+    /**
+     * Constructor
+     */
+    @Beta(value = Beta.SinceVersion.V4_18_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public BulkExecutionOptions() {
+        this(null);
+    }
+
+    /**
+     * Returns micro batch size
```

Comment:
To increase awareness of these APIs, in my opinion, getters should also have the same documentation as setters, except for the get and set part. 
Also, we should mention default values in the getters and setters.